### DISCUSSION TITLE
optimization chunkId/moduleId about named and deterministic

### DISCUSSION THREAD

**Discussion by lanmch** [webpack/webpack#15386] [Q&A]
As the docs, the config of optimization.chunkId/moduleId, we can set the value of 'deterministic' which is good for long term caching. And the value of 'named' is more readable.
In my project, we prepare to use the deterministic because of the term caching. But, when we want to find the error stack online, or find which module/file had included the bundle.js (link 125328.ajshdjkashd.js). The 'determistic' property seems like is difficult to find it.
I try to use the 'DeterministicChunkIdsPlugin' , but it looks like doesn't hav the relationship about ids and named. Also, I try to write a plugin, use the chunkIds/afterOptimizeChunks/optimizeChunkIds to get the relationship, but the callback params 'chunks' also doesn't have the relationship.
Could u help me? Or give me some advice?
I want to get the relationship of determisticId and name when I set the value  'determistic'.
thanks 🙏

**Comment by vankop** [comment]
You can use https://github.com/webpack/webpack/blob/main/lib/ids/SyncModuleIdsPlugin.js to store ids locally (it will look like a map where key is module identifier)

**Comment by lanmch** [comment]
i try this method, but it looks like moduleId with name rather than the chunkId with name

**Reply by vankop** [reply]
yeah, this about module only.. there is no way to do what you want without some "extra" code..

btw you probably just need source maps to map error stack trace.

**Reply by lanmch** [reply]
we hav source map already, but someone in our teams want to hav this feature. Maybe sometime the source map work not so good as we expect

**Reply by lanmch** [reply]
maybe i think i should write the plugin 'SyncChunkIdsPlugin' like the 'SyncModuleIdsPlugin'. 
Does the webpack team have any recent plans to support this feature? If not, i try it by myself

**Reply by vankop** [reply]
no, I think you should use source maps

**Reply by lanmch** [reply]
but if there is no error, I just want to find the 127362.sadasd.js is import by which file, the sourcemap looks like can't solve it

**Reply by vankop** [reply]
yeah, this is not possible. but why you need this? `webpack` is kind of tool to create you bundle just work.. (if you setup all well)

**Reply by vankop** [reply]
statoscope maybe get some info from stats https://statoscope.tech/ check it out

**Reply by lanmch** [reply]
> yeah, this is not possible. but why you need this? `webpack` is kind of tool to create you bundle just work.. (if you setup all well)

some time we use the dynamic import by 'react-loadable', so sometime some dynamic js file will be load when we open a website.
But, when we want to know which file import these dynamic js (like 127362.sadasd.js) we find it is very difficult. Maybe in my team, we want to decrease the unnessary js loading to impove out site quickly.
